sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -134,9 +132,6 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
       AlterDatabasePropertiesCommand(db, properties)
 
     case SetNamespaceLocation(DatabaseInSessionCatalog(db), location) if conf.useV1Command =>
-      if (StringUtils.isEmpty(location)) {
```

Comment:
I'm a bit worried about the perf impact here, shouldn't we always check empty string first then try creating URI?

Reply:
It won't create URI for null and empty string

Reply:
do you mean `new Path(str).toUri` already do fail-fast for empty or null string?

Reply:
Yes, in Path's constructor

Reply:
I'm not very comfortable relying on external library's implementation. Can't we do an explicit empty/null check at the Spark side? Just like the old code did.

Reply:
We extensively use `new Path` in the Spark repository. And here we  rely on its `toUri` for the entire string parsing. It's odd to distrust it for these simple steps.